NIMATRO/NIMATRO-IOC-01App/Db/controls.db

```diff
@@ -0,0 +1,422 @@
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:CONTROL:STOP")
+    field(SDIS, "$(P)DISABLE")
+}
+
+# Write to LabVIEW control "Mode" on "C:/LabVIEW Modules/Drivers/Nima Trough/Nima Trough Code/VIs/Nima Trough Interface.vi"
+record(mbbo, "$(P)MODE")
+{
+	field(DESC, "LabVIEW 'Mode'")
+    field(DTYP, "asynInt32")
+    field(OUT,  "@asyn(lvfp,0,0)Mode")
+
+    field(ZRVL, 0)
+    field(ZRST, "Barr Speed Control")
+        
+    field(ONVL, 1)
+    field(ONST, "Pressure Control")
+        
+    field(TWVL, 2)
+    field(TWST, "Area Control")
+        
+    field(THVL, 3)
+    field(THST, "Isotherm-Cycles")
+        
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:MODE")
+    field(SDIS, "$(P)DISABLE")
+}
+
+
+
+
+
+# Read LabVIEW control/indicator "Return Message" on "C:/LabVIEW Modules/Drivers/Nima Trough/Nima Trough Code/VIs/Nima Trough Interface.vi"
+record(waveform, "$(P)RETURN:MESSAGE")
+{
+	field(DESC, "LabVIEW 'Return Message'")
+    field(DTYP, "asynOctetRead")
+	field(FTVL, "CHAR")
+	field(NELM, 256)
+    field(INP,  "@asyn(lvfp,0,0)Return_Message")
+    field(SCAN, ".5 second")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:RETURN:MESSAGE")
+    field(SDIS, "$(P)DISABLE")
+}
+
+# Read LabVIEW control/indicator "Target Reached" on "C:/LabVIEW Modules/Drivers/Nima Trough/Nima Trough Code/VIs/Nima Trough Interface.vi"
+record(bi, "$(P)TARGET:REACHED")
+{
+	field(DESC, "LabVIEW 'Target Reached'")
+    field(DTYP, "asynInt32")
+    field(INP,  "@asyn(lvfp,0,0)Target_Reached")
+    field(SCAN, ".5 second")
+    field(ZNAM, "NO")
+    field(ONAM, "YES")
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:TARGET:REACHED")
+    field(SDIS, "$(P)DISABLE")
+}
+
+# Write to LabVIEW control "Close Barrier" on "C:/LabVIEW Modules/Drivers/Nima Trough/Nima Trough Code/VIs/Nima Trough Interface.vi"
+record(bo, "$(P)BARRIER:CLOSE")
+{
+	field(DESC, "LabVIEW 'Close Barrier'")
+    field(DTYP, "asynInt32")
+    field(OUT,  "@asyn(lvfp,0,0)Close_Barrier")
+    field(ZNAM, "")
+    field(ONAM, "CLOSE")
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:BARRIER:CLOSE")
+    field(SDIS, "$(P)DISABLE")
+}
+
+# Write to LabVIEW control "Open Barrier" on "C:/LabVIEW Modules/Drivers/Nima Trough/Nima Trough Code/VIs/Nima Trough Interface.vi"
+record(bo, "$(P)BARRIER:OPEN")
+{
+	field(DESC, "LabVIEW 'Open Barrier'")
+    field(DTYP, "asynInt32")
+    field(OUT,  "@asyn(lvfp,0,0)Open_Barrier")
+    field(ZNAM, "")
+    field(ONAM, "OPEN")
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:BARRIER:OPEN")
+    field(SDIS, "$(P)DISABLE")
+}
+
+# Write to LabVIEW control "Stop Barrier" on "C:/LabVIEW Modules/Drivers/Nima Trough/Nima Trough Code/VIs/Nima Trough Interface.vi"
+record(bo, "$(P)BARRIER:STOP")
+{
+	field(DESC, "LabVIEW 'Stop Barrier'")
+    field(DTYP, "asynInt32")
+    field(OUT,  "@asyn(lvfp,0,0)Stop_Barrier")
+    field(ZNAM, "")
```

Comment:
Give this a value. No further comments about this - please check all records

Reply:
Values have been included for both input string names